QueryBuilder/JoinQuery.cs

```diff
@@ -55,7 +55,7 @@ internal JoinQuery(IDictionary<string, Type> aliasToTypeMapping, SelectClause se
         {
             var relationship = relationshipSelector.Invoke((TJoinFrom)Activator.CreateInstance(typeof(TJoinFrom)));
 
-            if (!aliasToTypeMapping.ContainsKey(joinFromAlias))
+            if (string.IsNullOrEmpty(joinFromAlias) || !aliasToTypeMapping.ContainsKey(joinFromAlias))
```

Comment:
Rather than check for `null` here, we should avoid having `null` alias values reach this point in the first place. There should be some code further up that manages resolving the alias (check `ValidateAndGetAlias`) and throws an exception if no alias exists.

I'm also not sure why we check that `aliasToTypeMapping` contains the alias here - shouldn't `ValidateAndGetAlias` already handle that?

Reply:
> I'm also not sure why we check that aliasToTypeMapping contains the alias here - shouldn't ValidateAndGetAlias already handle that?

This is a first-class method, and the user can call it directly and pass an undefined "from" alias

Reply:
@Isai-Cicourel in what scenario is it possible to reach a `null` joinFromAlias?

Reply:
Removed the null validation as per our discussion